### ISSUE TITLE
which version for tensorflow is the best?

### ISSUE THREAD

**Issue by quanheng123** [matterport/Mask_RCNN#1808]
Hi, new here.
just want to know which version is the best for the project.
because Ive changed so many version from 1.0 to 2,0 and most of the time there is the bug for no module for **** or attribute error for TensorFlow.

thanks.

**Comment by mangalbhaskar** [comment]
Hope this may help, I'm running successfully on the following versions for different scenarios:

Compatibility for **Prediction**:
- tensorflow-gpu==1.9.0,  keras==2.2.2, CUDA 9.0 - this is the best
- tensorflow-gpu==1.13.1, keras==2.2.3, CUDA 10.0
- tensorflow-gpu==1.14.0 with CUDA 10.0

Compatibility for **Training**:
- tensorflow-gpu==1.9.0,  keras==2.2.2, CUDA 9.0 - this is the best
- tensorflow-gpu==1.13.1, keras==2.2.3, CUDA 10.0

**Comment by subbulakshmisubha** [comment]
@mangalbhaskar so tensorflow version 1.14.0 with cuda 10 would be compatible with which version of tensorflow gpu?

**Comment by mangalbhaskar** [comment]
@subbulakshmisubha tf-1.14.0 requires keras-2.3.0 and CUDA 10; but the code is not compatible with keras 2.3.0. You can use: tensorflow-1.13.1, keras-2.2.3 with CUDA 10.0
Thanks to @saqibmobin to discover the compatibility with the CUDA-10

**Comment by YubinXie** [comment]
> Hope this may help, I'm running successfully on the following versions for different scenarios:
> 
> Compatibility for **Prediction**:
> 
> * tensorflow-gpu==1.9.0,  keras==2.2.2, CUDA 9.0 - this is the best
> * tensorflow-gpu==1.13.1, keras==2.2.3, CUDA 10.0
> * tensorflow-gpu==1.14.0 with CUDA 10.0
> 
> Compatibility for **Training**:
> 
> * tensorflow-1.9.0,  keras-2.2.2, CUDA 9.0 - this is the best
> * tensorflow-1.13.1, keras-2.2.3, CUDA 10.0

I am curious that why for training, you did not put GPU in the TensorFlow version? 
Thanks for sharing!

**Comment by mangalbhaskar** [comment]
@YubinXie it was the miss-out in typing; I've fixed the typo with the tf gpu version

**Comment by YubinXie** [comment]
> @YubinXie it was the miss-out in typing; I've fixed the typo with the tf gpu version

I see. I really appreciate you sharing the version validation. It is really helpful!

**Comment by HungPhanbku** [comment]
Is Mask RCNN support for cuda 10.2 using GPU and which version tensorflow and keras? thanks

**Comment by mangalbhaskar** [comment]
@HungPhanbku - I don't think so because this main repo is not updated for long now. For CUDA 10.2 support, you need to have TF 2.x; and in TF2.x keras is in-built and then this code probably not compatible with TF 2.X. But, check some forks and pull requests; people have tried migrating this for TF 2.X.

**Comment by Shakesbeer333** [comment]
@HungPhanbku did you find a solution for Cuda 10.2 (and TF 2.x)?

**Comment by HungPhanbku** [comment]
tensorflow-gpu=1.14.0, keras=2.2.3, python 3.7 make sure export PATH,CUDADIR, LD_LIBRARY_PATH
It's work okay

**Comment by Samkhan705279** [comment]
2.9.0 is most stable i had found till now 21-07-2023

**Comment by rhuanbarros** [comment]
I'm using this docker image (tensorflow/tensorflow:2.15.0-gpu-jupyter) and it's working fine. The latest version (2.16) has a lot of bugs in relation to the new keras 3 and tensorboard.
